RARLab.WinRAR version 6.20.0

borque opened the pull request:
I have stripped some languages from this manifest and also not all previous ones are available for WinRAR 6.20.0. I presume they will be re-added in 6.21.0, which is currently in Beta and will try to provide a fully featured manifest-update then.

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.4 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.4.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/96243)

borque committed:
RARLab.WinRAR version 6.20.0

borque committed:
Update RARLab.WinRAR.locale.de.yaml

Trenly reviewed (changes requested):
Please use the BCP-47 language codes. `en`, `sv`, `it`, `de`, and `fr` are not compliant.

borque commented:
Hello!
Thanks for reviewing and the suggestions.

Concerning the language codes: I have seen some manifests here which only use the first part of a language declaration, if only one version exists for various „dialects“. E.g. DE exists with -de for Germany, -AT for Austria and -CH for Switzerland, but WinRAR only knows one package for that. As different currencies or other (minor) differences are of no relevance for archiving, there is no problem for RARlab in having just a single package, I presume. So I followed other examples and only declared the first part and as it passed local and pipeline validations and selected the correct package on my local machine I believed it was okay.

Which locale should I choose in such cases to be compliant to standard? And if I declare de-DE will systems running de-AT see de-DE because of the common „root“ de- or will they get the default locale en-US?

Let me state these are only practical considerations and I don‘t want to offend or humiliate people speaking different variants of a language with that.

Concerning the missing languages: es stated in the initial post of my PR-submission, some of the languages present in the old manifest are missing in WinRARs 6.20.0-installers, some others were added. As 6.20.0 was followed by a beta for a bugfixing 6.21.0-release with more languages relatively fast, I considered this to be a kind of short-lived interim release and did not bother to implement or remove all languages as they will be changing again in the near future.

Trenly commented:
Hmm. I was of the understanding that if the full tag wasn’t used, it could cause issues with locale selection; especially if users have a preference set in their settings file

Perhaps @denelon can provide guidance on what is best here

denelon commented:
The two-part <country>-<language> BCP-47 codes are best. We have some heuristics to try and determine which locale is the best to use when there isn't an exact match. I'll ask one of the engineers to give a bit more detail on Monday.

borque committed:
Update RARLab.WinRAR.installer.yaml

Changed language-codes according to discussion

borque commented:
**@Trenly** : 
I have changed the language-codes according to your request.
**@denelon** : 
Did you get any feedback from the engineers? 
If codes according to XX-yy should always be used here, how to proceed if only a single installer exists for XX? 
Does not matter which yy I choose in the manifest then? To stay with my example: thre is only a DE-Installer, but there are three codes for it: de-DE (Germany), de-AT (Austria), de-CH (Switzerland). Will all these countries get the German-language installer if I encode it de-DE instead of only de in the manifest

ItzLevvie commented on manifests/r/RARLab/WinRAR/6.20.0/RARLab.WinRAR.installer.yaml:
```diff
@@ -0,0 +1,95 @@
+  Architecture: x86
+  InstallerUrl: https://www.rarlab.com/rar/winrar-x32-620es.exe
+  InstallerSha256: 109CEA214AC974C8AD86CCB90117701AF177FC62B3ED51F20676278644767B13
+- InstallerLocale: sv-SV
```
```suggestion
- InstallerLocale: sv-SE
```

ItzLevvie commented on manifests/r/RARLab/WinRAR/6.20.0/RARLab.WinRAR.installer.yaml:
```diff
@@ -0,0 +1,95 @@
+  Architecture: x64
+  InstallerUrl: https://www.rarlab.com/rar/winrar-x64-620sw.exe
+  InstallerSha256: CDE7A5924094D49C7E6E1E30239C03771753C878E324D89593B5BEC138D0654F
+- InstallerLocale: sv-SV
```
```suggestion
- InstallerLocale: sv-SE
```

borque replied:
Sometimes it's not as easy as you hope... I thought first and second letterpair of the language descriptor are identical... Learned and noted for future manifests!

borque committed:
Update manifests/r/RARLab/WinRAR/6.20.0/RARLab.WinRAR.installer.yaml

Sometimes it's not as easy as you hope... I thought first and second letterpair of the language descriptor are identical... Learned and noted for future manifests!

Co-authored-by: Levvie - she/her <11600822+ItzLevvie@users.noreply.github.com>

ItzLevvie commented on manifests/r/RARLab/WinRAR/6.20.0/RARLab.WinRAR.installer.yaml:
```diff
@@ -0,0 +1,95 @@
+  Architecture: x64
+  InstallerUrl: https://www.rarlab.com/rar/winrar-x64-620sw.exe
+  InstallerSha256: CDE7A5924094D49C7E6E1E30239C03771753C878E324D89593B5BEC138D0654F
+- InstallerLocale: sv-SV
```
```suggestion
- InstallerLocale: sv-SE
```
Looks like you forgot to fix the locale for the x64 version

borque committed:
Update RARLab.WinRAR.installer.yaml

borque commented:
@ItzLevvie : My apologies...
